db/db_impl/db_impl.cc

```diff
@@ -548,12 +548,39 @@ Status DBImpl::CloseHelper() {
   flush_scheduler_.Clear();
   trim_history_scheduler_.Clear();
 
+  // For now, simply trigger a manual flush at close time
+  // on all the column families.
+  if (immutable_db_options_.experimental_allow_mempurge) {
```

Comment:
At the moment, automagically flush all column families.  In the future I can do a more fine-grained flushing by first checking if there is a need for flushing (but need to implement something else than imm()->IsFlushPending() because the output memtables added to imm() dont trigger flushes).